src/DynamicRupture/FrictionLaws/GpuImpl/FastVelocityWeakeningLaw.h

```diff
@@ -0,0 +1,150 @@
+#ifndef SEISSOL_GPU_RATEANDSTATEFASTVELOCITYWEAKENING_H
+#define SEISSOL_GPU_RATEANDSTATEFASTVELOCITYWEAKENING_H
+
+#include "DynamicRupture/FrictionLaws/GpuImpl/RateAndState.h"
+
+namespace seissol::dr::friction_law::gpu {
+
+template <typename TPMethod>
+class FastVelocityWeakeningLaw
+    : public RateAndStateBase<FastVelocityWeakeningLaw<TPMethod>, TPMethod> {
+  public:
+  using RateAndStateBase<FastVelocityWeakeningLaw, TPMethod>::RateAndStateBase;
+
+  void copyLtsTreeToLocal(seissol::initializers::Layer& layerData,
+                          seissol::initializers::DynamicRupture const* const dynRup,
+                          real fullUpdateTime) {}
+
+  void copySpecificLtsDataTreeToLocal(seissol::initializers::Layer& layerData,
+                                      seissol::initializers::DynamicRupture const* const dynRup,
+                                      real fullUpdateTime) override {
+    using SelfInitializerType = seissol::initializers::LTSRateAndStateFastVelocityWeakening;
+    auto* concreteLts = dynamic_cast<SelfInitializerType const* const>(dynRup);
+    this->srW = layerData.var(concreteLts->rsSrW);
+
+    using ParentType = RateAndStateBase<FastVelocityWeakeningLaw<TPMethod>, TPMethod>;
+    ParentType::copySpecificLtsDataTreeToLocal(layerData, dynRup, fullUpdateTime);
+  }
+
+  struct Details {
+    decltype(FastVelocityWeakeningLaw::a) a;
+    decltype(FastVelocityWeakeningLaw::sl0) sl0;
+    decltype(FastVelocityWeakeningLaw::srW) srW;
+    decltype(dr::DRParameters::rsSr0) rsSr0;
+    decltype(dr::DRParameters::rsF0) rsF0;
+    decltype(dr::DRParameters::rsB) rsB;
+  };
+
+  Details getCurrentLtsLayerDetails() {
+    Details details{};
+    details.a = this->a;
+    details.sl0 = this->sl0;
+    details.srW = this->srW;
+    details.rsSr0 = this->drParameters->rsSr0;
+    details.rsF0 = this->drParameters->rsF0;
+    details.rsB = this->drParameters->rsB;
+    return details;
+  }
+
+  void updateStateVariable(double timeIncrement) {
```

Comment:
Why isn't there an `impl` version of this function?

Reply:
I guess it is not relevant anymore because I renamed `updateStateVariableImpl` to `updateStateVariable`

done